routes/auth.routes.js

```diff
@@ -0,0 +1,103 @@
+const { Router } = require("express");
+const bcrypt = require("bcrypt");
+const passport = require("passport");
+const router = Router();
+const saltRounds = 10;
+const User = require("../models/User.model");
+const { Mongoose } = require("mongoose");
+const mongoose = require("mongoose")
+
+router.get("/signup", (req,res,next) => res.render("auth/signup"));
+
+router.post("/signup", (req, res) => {
+    const { username, password } = req.body;
+
+    if (!username || !password){
+        if (password !== /^\S+@\S+\.\S+$/) {
+            res.render("auth/signup", {
+                validationError: "Please user a valid email adress.",
+                errorMessage:
+                "All fields are mandatory. Please provide your username, email and password.",
+            });
+        }
+        return;
+    }
```

Comment:
Nice work on the validation. I just wanted to leave out a tip on how you can make this piece of code a bit more readable while maintaining the functionality. 

You can split up what the validation does into two things. I know, I did it this way in class, but this will also work:

```suggestion
    // Validate empty fields
    if (!username || !password){
            res.status(500).render("auth/signup", {
      errorMessage:
        "All fields are mandatory. Please provide your username, email and password.",
    });
    return;
    }
    
    // Validate email
    const emailFormatRegex = /^\S+@\S+\.\S+$/;
    
    if (!emailFormatRegex.test(email)) {
    res.status(500).render("auth/signup", {
      validationError: "Please use a valid email address.",
    });
    return;
  }
```